cypress-io/cypress
# fix: remove CT side effects from mount when e2e testing

- Closes #22589

### User facing changelog
CT side effects from `import { mount } from 'cypress/{react,vue}'` are disabled when running e2e tests.

### Additional details
Though we scaffold CT with the mount registration in `cypress/support/component.js`, a user might lift this registration to the scaffolded `commands.js` file, which is not specific to testing type. There are side effects associated with the mount import that are required for CT testing but they will cause errors when e2e testing.

Though we have e2e and component specific support files, users might be inclined to keep all of their custom commands in a single file as documented [here](https://docs.cypress.io/api/cypress-api/custom-commands). Doing so should not cause these errors to surface, so I've stubbed out the CT side effects when running e2e (and improved a few errors thrown due to not finding the root selector).

Ideally, I don't want the `mount` import to have side effects. I think there should be an explicit import/registration that would be used inside of the `component.js`support file, but that would be a breaking change so I've left some comments around a future refactor of this.

### Steps to test
The original issue has a good reproduction and I've included a system test to verify the new behavior. You can scaffold a basic project with e2e and component testing and lift the `Cypress.Commands.add('mount', ...)` registration into `commands.js` then try to run an e2e test and the error will be thrown.

### How has the user experience changed?
Before: 
![Screen Shot 2022-06-30 at 2 14 00 PM](https://user-images.githubusercontent.com/25158820/176759326-45f8f6df-95e3-447b-86e4-77d829c18191.png)

After:
![Screen Shot 2022-06-30 at 2 09 08 PM](https://user-images.githubusercontent.com/25158820/176759366-45fef78b-5274-4916-9713-d4c4fa970174.png)


### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
